pkg/storage/filesystem/device.go

```diff
@@ -83,6 +95,83 @@ func (d *DeviceInfo) IsPXEPartition() bool {
 	return d.Label == "ZOSPXE"
 }
 
+func (d *DeviceInfo) IsPartitioned() bool {
+	return len(d.Children) != 0
+}
+
+func (d *DeviceInfo) GetUnallocatedSpaces(ctx context.Context) ([]DiskSpace, error) {
+	args := []string{
+		"--json", d.Path, "unit", "B", "print", "free",
+	}
+	output, err := exec.CommandContext(ctx, "parted", args...).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("parted command error: %w", err)
+	}
+
+	var diskData struct {
+		Disk struct {
+			Partitions []DiskSpace `json:"partitions"`
+		} `json:"disk"`
+	}
+	if err := json.Unmarshal(output, &diskData); err != nil {
+		return nil, fmt.Errorf("failed to parse parted output: %v", err)
+	}
+
+	validSpaces := []DiskSpace{}
+	for _, part := range diskData.Disk.Partitions {
+		if isValidAsDevice(part) {
+			validSpaces = append(validSpaces, part)
+		}
+	}
+
+	return validSpaces, nil
+}
+
+func (d *DeviceInfo) AllocateEmptySpace(ctx context.Context, space DiskSpace) error {
+	args := []string{
+		d.Path, "mkpart", "primary", space.Start, space.End,
+	}
+
+	output, err := exec.CommandContext(ctx, "parted", args...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("parted command error: %w", err)
+	}
+	log.Debug().Str("output", string(output)).Msg("allocate empty space parted command")
+
+	return nil
+}
+
+func (d *DeviceInfo) RefreshDeviceInfo(ctx context.Context) (DeviceInfo, error) {
+
+	// notify the kernel with the changed
+	if err := Partprobe(ctx); err != nil {
+		return DeviceInfo{}, err
+	}
+
+	// remove the cache
+	d.mgr.ClearCache()
+
+	return d.mgr.Device(ctx, d.Path)
+}
+
+func isValidAsDevice(space DiskSpace) bool {
+	// empty space around and in-between partitions used for
+	// partitioning table metadata and partitions alignment
+	const minDeviceSizeBytes = 2 * 1024 * 1024 // 2 MB
```

Comment:
I think a 2MB disk is not really usable in practice. I suggest this is in creased to 2GB or even 5GB 

What do you think @xmonader ?